packages/components/src/tabs/index.tsx

```diff
@@ -154,26 +159,49 @@ function Tabs( {
 		setSelectedId,
 	] );
 
-	// In controlled mode, make sure browser focus follows the selected tab if
-	// the selection is changed while a tab is already being focused.
-	useLayoutEffect( () => {
-		if ( ! isControlled || ! selectOnMove ) {
+	const previousSelectedId = usePrevious( selectedId );
+
+	useEffect( () => {
```

Comment:
Correct! As a `useLayoutEffect` it got caught up in the other layout changes the component calls. In some cases this would mean focus would get updated twice, the second time throwing it out of sync with the currently selected tab. Specifically in the editor, this happened when tabbing from the last block on the canvas over into the settings sidebar.